activerecord/lib/rails/generators/active_record/model/model_generator.rb

```diff
@@ -43,9 +43,16 @@ def accessible_attributes
 
         # Used by the migration template to determine the parent name of the model
         def parent_class_name
-          options[:parent] || "ActiveRecord::Base"
+          options[:parent] || determine_default_parent_class
         end
 
+        def determine_default_parent_class
+          if File.exist?('app/models/application_record.rb')
```

Comment:
Is it safe to expect that the app root is the current working directory here, or should it check for the file presence relative to application root?